text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,641 @@
+multiple messages (in case messages are passed for processing as batches), or
+for no message at all (if it is signalled that no messages were received).  For
+each message it accounts for, the "Deliver" or "Receive" span SHOULD link to
+the "Create" or "Publish" span for the message. In addition, if it is possible
+the "Create" or "Publish" span MAY be set as a parent of the "Deliver" or
+"Receive" span.
+
+#### Settlement spans
+
+"Settle" spans SHOULD be created for every manually or automatically triggered
+settlement operation. A single "Settle" span can account for a single message
+or for multiple messages (in case messages are passed for settling as batches).
+For each message it accounts for, the "Settle" span MAY link to the "Create" or
+"Publish" span of the message.
+
+## Open issues
+
+Fully integrating the changes proposed in this document into messaging semantic
+conventions requires some additions and clarifications in the specification,
+which are listed in this section.
+
+* [open-telemetry/opentelemetry-specification#454](https://github.com/open-telemetry/opentelemetry-specification/issues/454)
+  To instrument push-based "Receive" operations as described in this document,
+  it is necessary to add links to spans after those spans were created. The
+  reason for this is, that not all messages are present at the start of a
+  "Receive" operations, so links to related contexts cannot be added at the start
+  of the span.
+* [open-telemetry/opentelemetry-specification#2176](https://github.com/open-telemetry/opentelemetry-specification/issues/2176)
+  When consuming a message with no attached creation context as part of a
+  batch, it would still be useful to capture related message-specific
+  attributes as part of a link which points to an invalid context. However,
+  according to the specification links pointing to an invalid context may be
+  ignored. For consistently reporting message-specific attributes on links, links
+  to invalid contexts should be allowed and supported.
+* [open-telemetry/opentelemetry-specification#3172](https://github.com/open-telemetry/opentelemetry-specification/issues/3172)
+  Currently the specification is unclear about whether relationships between
+  producer and consumer spans can be modelled via links, the wording suggests
+  that it should be a parent/child relationship. The wording in the specification
+  needs to make it clear that this can be a link too.
+
+## Examples
+
+This section contains a list of examples illustrating the use of the
+conventions outlined above. Green boxes denote spans that are required to
+exist in order to conform to those conventions. Other boxes denote spans that
+are not required and covered by the conventions, but are hopefully helpful in
+understanding how messaging spans can be integrated into an overall trace flow.
+Solid arrows denote parent/child relationships, dotted arrows denote link
+relationships.
```

Comment:
I find the direction of the link arrows confusing. While this text doesn't specifically say which span has the link, I would think it is clearest for the base of the arrow to be connected to the span that has the link and the head of the arrow to point at the span the link is referencing. If other find the reverse to be more natural, might it help to define which end of the arrow has the link and which end the link references?

Reply:
I updated diagrams now to just use dotted lines without arrows for links.